react-app/src/History.js

```diff
@@ -65,117 +66,164 @@ export default function History({ db, reset }) {
-                      'just for testing',
-                      randomTimeInMarch,
-                    ])
-                  })
+      <View style={styles.container}>
+        <View style={styles.calendarContainer}>
+          <View style={styles.calendar}>
+            <Calendar
+              markedDates={{ ...records, ...selected }}
+              onDayPress={(day) => {
+                const localTimestamp = day.timestamp + new Date().getTimezoneOffset() * 60 * 1000
+                setSelected({
+                  [day.dateString]: {
+                    selected: true,
+                    dotColor: 'blue',
+                    selectedColor: '#FFCD28',
+                  },
                 })
-              db.transaction((tx) => {
-                tx.executeSql('select * from Record', [], (_, { rows }) => {
-                  let result = rows._array.reduce(
-                    (previous, current) => ({
-                      ...previous,
-                      [toYYYYMMDD(current.timestamp)]: { marked: true },
-                    }),
-                    {}
+                db.transaction((tx) => {
+                  tx.executeSql(
+                    'select timestamp, exercise_name from Record where timestamp between ? and ? order by timestamp',
+                    [localTimestamp, localTimestamp + 24 * 60 * 60 * 1000 - 1],
+                    (_, { rows }) => {
+                      setRecordsOfSelected(rows._array)
+                    }
                   )
-                  setRecords(result)
                 })
-              })
-            }}
-          />
-        )}
-        {__DEV__ && (
-          <Button
-            title='console log'
-            color='#005408'
-            onPress={() => {
-              console.log(recordsOfSelected)
-            }}
+              }}
+              onMonthChange={() => {
+                setRecordsOfSelected([])
+              }}
+              hideExtraDays={true}
+              theme={{
+                calendarBackground: ThemeColor.backgroundColor,
+                dayTextColor: ThemeColor.textColor,
+                arrowColor: ThemeColor.textColor,
+                monthTextColor: ThemeColor.textColor,
+              }}
+            />
+          </View>
+          <Text style={styles.textTitle}>Exercise Record</Text>
+        </View>
+        {/* Text of the record */}
+        <View>
+          <FlatList
+            data={recordsOfSelected}
+            renderItem={({ item }) => (
+              <View style={styles.recordContainer}>
+                <View style={styles.recordInfo}>
+                  <Text style={styles.exerciseName}>{item.exercise_name}</Text>
+                  <Text style={styles.timeStamp}>{formatDate(item.timestamp)}</Text>
+                </View>
+                <Text style={styles.exerciseDetail}>
+                  {STRETCHING_EXERCISE_DATA.length} Moves, 10 minutes
```

Comment:
This part is challenging one for me.
I attempted some ways but I could't get data with 'exercise_name'...
i hope someone help me.